nemo_skills/code_execution/sandbox.py

```diff
@@ -372,42 +374,63 @@ def update_fn(map_to_future, line_dict, answer_format="natural_language"):
                     if answer_format == "natural_language":
                         gt_answer = line_dict["expected_answer"]
 
-                    if not use_predicted_answer_key:
-                        if answer_format == "natural_language":
+                    if answer_format == "natural_language":
+                        if not use_predicted_answer_key:
                             line_dict["predicted_answer"] = extract_answer(
                                 line_dict["generation"],
                                 extract_from_boxed=extract_from_boxed,
                                 extract_regex=extract_regex,
                             )
-                        elif answer_format == "lean":
-                            line_dict["predicted_answer"] = (
-                                line_dict["header"]
-                                + line_dict["formal_statement"]
-                                + (
-                                    line_dict["generation"][:-3]
-                                    if line_dict["generation"].endswith("```")
-                                    else line_dict["generation"]
+                        else:
+                            if "predicted_answer" not in line_dict:
+                                raise ValueError(
+                                    "predicted_answer key not found in the line_dict. "
+                                    "Set use_predicted_answer_key=False to re-extract"
                                 )
-                            )
-                    else:
-                        if "predicted_answer" not in line_dict:
-                            raise ValueError(
-                                "predicted_answer key not found in the line_dict. "
-                                "Set use_predicted_answer_key=False to re-extract"
+                    # TODO manually removing <｜begin▁of▁sentence｜> for trtllm
+                    elif answer_format == "lean":
+                        if not use_predicted_proof_key:
+                            generation = re.sub(r"^```(lean4)?\s*|\s*```$", "", line_dict["generation"])
+                            line_dict["predicted_proof"] = (
+                                line_dict["header"] + line_dict["formal_statement"] + generation
                             )
 
+                        else:
+                            if "predicted_proof" not in line_dict:
+                                raise ValueError(
+                                    "predicted_proof key not found in the line_dict. "
+                                    "Set use_predicted_proof_key=False to re-combine"
+                                )
+                    elif answer_format == "lean-stat":
```

Comment:
let's call it `lean-statement` (if that's what it stands for), as stat is ambiguous

Reply:
and maybe the other one rename to `lean-proof` to make it clearer

Reply:
and probably add 4, since you use it in other places. So `lean4-statement` and `lean4-proof`

Reply:
Done